src/coreclr/jit/abi.cpp

```diff
@@ -287,6 +287,27 @@ ABIPassingInformation ABIPassingInformation::FromSegment(Compiler* comp, const A
     return info;
 }
 
+//-----------------------------------------------------------------------------
+// FromSegments:
+//   Create ABIPassingInformation from multiple segments.
+//
+// Parameters:
+//   comp    - Compiler instance
+//   segments - Segments that represents the passing information
+//
+// Return Value:
+//   An instance of ABIPassingInformation.
+//
+ABIPassingInformation ABIPassingInformation::FromSegments(Compiler*                                comp,
+                                                          std::initializer_list<ABIPassingSegment> segments)
```

Comment:
Does `std::initializer_list` guarantee no heap allocation and no possibility of C++ exceptions? It's very new that we're able/allowed to use STL in the JIT, and there are restrictions being documented in #101088. 

It might be better to use a variadic template here if we're unsure. There are several other places that we can switch to `std::initializer_list` if we are confident that we can use it.

Reply:
The [constructor is noexcept](https://en.cppreference.com/w/cpp/utility/initializer_list/initializer_list). Pretty sure it also doesn't heap allocate as it doesn't take any custom allocators like anything which allocates in STL but apparently the [guarantee of underlying array being a temporary array of type const T\[N\] starts since C++14](https://stackoverflow.com/questions/61825532/does-stdinitializer-list-heap-allocate-memory). So maybe let's err on the safe side.

There's a problem with variadic template, the body in abi.h would need operator new from compiler.h but abi.h is included in compiler.h (moving the include down like "loopcloning.h" causes problems). I made a manual allocation like on other archs.